pex/resolve/locker_patches.py

```diff
@@ -154,6 +154,11 @@ def supported_platform_tag(self, *_args, **_kwargs):
         check(*args, **kwargs) for check in supported_checks
     )
 
+    # N.B.: This patch is a noop for the 20.3.4-patched Pip but is required in newer Pip.
```

Comment:
This was the only change actually required to get Pip 22.2.2 working. The PEP-691 work was just needed to make locking comparably fast with vendored Pip (without PEP-691 each artifact is downloaded just so it can be hashed locally using 22.2.2).